src/curl.c

```diff
@@ -64,6 +64,27 @@ static void swupd_curl_strerror(CURLcode curl_ret)
 	error("Curl - Download error - (%d) %s\n", curl_ret, curl_easy_strerror(curl_ret));
 }
 
+static char spinner[5] = { '|', '/', '-', '\\', '\\' };
+
+static int progress_callback(void UNUSED_PARAM *clientp, double UNUSED_PARAM dltotal,
+			     double UNUSED_PARAM dlnow, double UNUSED_PARAM ultotal, UNUSED_PARAM double ulnow)
+{
+	static int counter;
+	if (counter % 100000000 == 0) {
+		static int index;
+		info("[%c]", spinner[index % 6]);
+		fflush(stdout);
+		index++;
+		info("\b\b\b");
```

Comment:
Good point!

Reply:
There is a slight problem with that:

When I do that, i notice a cursor move and flicker which I dont see in case of \b\b\b. Not sure why that happens but. I am checking why

Reply:
@castulo Not sure why, but the cursor keeps flickering.

Reply:
Visually for me the best is this:

		info("\r");
		info("[%c]", spinner[index]);
 		fflush(stdout); 

Because the cursor will be after the message and not before.

Reply:
I think that i like that idea..because even if user types something by mistake, the bar does not overwritten.

Reply:
Done